### ISSUE TITLE
Legrövidebb út megtalálása egy adott csúcshoz

### ISSUE THREAD

**Issue by Abrisss** [szentkiralyi91/RouteSearch#9]
Ha kiválasztunk egy várost, akkor az ahhoz legközelebb eső, tranzit táblában levő városhoz az utat meg kell találni.

**Comment by Abrisss** [comment]
Ötlet:
Megnézzük, hogy a kiindulási csúcson milyen utak mennek át, majd ezen utak kezdő és végpontját lekérjük. Majd a célcsúcs és ezen végpontok között meghatározzuk a távolságot. És azon az úton megyünk tovább, ahol ez a távolság minimális. Amelyik pontnál minimális volt a távolság az lesz a következő kiindulási pont.

**Comment by kukovecz** [comment]
Igen, ez sem lenne egy rossz ötlet. Az elmúlt pár napban már elég sokféleképp nekifutottam az útkeresésnek, de sajnos nem a legnagyobb sikerekkel.

Röviden vázolva a probléma (bár ez akkora baj szerintem hogy legalább 5 issue-t megérne :)  
A shp fájlban az "adatszerkezetek" nagyon nehezen használhatóak. Van két külön réteg, egyik csak a városoknak a másik pedig csak a utakért felelős. Ez még megoldható is lenne, mivel WKT-ként beolvasgatva a geometriai objektumokat azonosíthatóak a pontok és koordináták alapján matematikai összefüggésekkel meg lehet vizsgálni az egymáshoz képesti viszonyukat....
....viszont az utak adathalmaz ugye MINDEN utat (432 ezer +) rekordot tartalmaz. Ha csak a főutakra leszűrünk az is rengeteg, kvázi két szomszédos város között is legalább 100 út van....
Sőt, a legnagyobb probléma, hogy még a szomszédos utak is olykor diszjunknak, így nagyon nagyon nehezen tud ott a keresés továbbmenni...
